agent/api/task/task.go

```diff
@@ -322,6 +322,9 @@ func TaskFromACS(acsTask *ecsacs.Task, envelope *ecsacs.PayloadMessage) (*Task,
 	//initialize resources map for task
 	task.ResourcesMapUnsafe = make(map[string][]taskresource.TaskResource)
 
+        // initialize network mode so service connect validations are accurate
+        task.initNetworkMode()
+
 	// extract and validate attachments
 	if err := handleTaskAttachments(acsTask, task); err != nil {
```

Comment:
I think we also need to update `handleTaskAttachment` method to use the internal `task.networkMode` and NOT the one from ACS payload (the only difference for the internal one is that we'll fill out the empty aka default mode)

Reply:
https://github.com/aws/amazon-ecs-agent/blob/feature/service-connect/agent/api/task/task_attachment_handler.go#L62

Reply:
https://github.com/aws/amazon-ecs-agent/blob/feature/service-connect/agent/api/task/task.go#L356 also calls initNetworkMode(). Maybe we could remove it from PostUnmarshalTask method?

Reply:
its idempotent, so not super problematic as long as it gets called somewhere :)